packages/rpc-transport-http/src/http-transport-errors.ts

```diff
@@ -0,0 +1,16 @@
+type SolanaHttpErrorDetails = Readonly<{
+    statusCode: number;
+    message: string;
+}>;
+
+export class SolanaHttpError extends Error {
+    readonly statusCode: number;
+    constructor(details: SolanaHttpErrorDetails) {
+        super(`HTTP error (${details.statusCode}): ${details.message}`);
+        Error.captureStackTrace(this, this.constructor);
+        this.statusCode = details.statusCode;
+    }
+    get name() {
+        return 'SolanaHttpError';
+    }
+}
```

Comment:
Should these be here? Or should they be called `Solana`?

From PR description:

> Notice that it does not include any Solana-specific types

Reply:
I was going to replace that with the `SolanaError` from `@solana/errors` but we decided to do that outside of this PR so I just left it as-is for now.